platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/widgets/MyLocationView.java

```diff
@@ -865,6 +867,31 @@ public void onSensorChanged(SensorEvent event) {
       }
     }
 
+    /**
+     * Pulls out the rotation vector from a SensorEvent, with a maximum length
+     * vector of four elements to avoid potential compatibility issues.
+     *
+     * @param event the sensor event
+     * @return the events rotation vector, potentially truncated
+     */
+    @NonNull
+    float[] getRotationVectorFromSensorEvent(@NonNull SensorEvent event) {
+      if (event.values.length > 4) {
+        // On some Samsung devices SensorManager.getRotationMatrixFromVector
+        // appears to throw an exception if rotation vector has length > 4.
+        // For the purposes of this class the first 4 values of the
+        // rotation vector are sufficient (see crbug.com/335298 for details).
+        // Only affects Android 4.3
+        if (truncatedRotationVectorValue == null) {
```

Comment:
It seems that this condition is always `false` because we're initializing it here https://github.com/mapbox/mapbox-gl-native/blob/76d1d87cc1c54fbe72549810610159306555b729/platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/widgets/MyLocationView.java#L814 so it could be removed. Am I missing something?